File: backend/src/main/java/com/rakshapoorvak/model/entity/TriageRecord.java
package com.rakshapoorvak.model.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;

/**
 * Triage vitals record for an SOS event.
 */
@Entity
@Table(name = "triage_records")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TriageRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sos_event_id", nullable = false)
    private SosEvent sosEvent;

    @Column(name = "heart_rate")
    private Integer heartRate;

    @Column(name = "systolic_bp")
    private Integer systolicBp;

    @Column(name = "diastolic_bp")
    private Integer diastolicBp;

    @Column(name = "spo2")
    private Integer spo2;

    @Column(name = "temperature", precision = 4, scale = 2)
    private BigDecimal temperature;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;

    @Column(name = "recorded_at")
    private Instant recordedAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        if (createdAt == null) createdAt = Instant.now();
        if (recordedAt == null) recordedAt = Instant.now();
    }
}

File: driver-app/lib/data/models/triage_record.dart
class TriageRecord {
  final int? id;
  final int sosEventId;
  final int? heartRate;
  final int? systolicBp;
  final int? diastolicBp;
  final int? spo2;
  final double? temperature;
  final String? notes;
  final String? createdAt;

  const TriageRecord({
    this.id,
    required this.sosEventId,
    this.heartRate,
    this.systolicBp,
    this.diastolicBp,
    this.spo2,
    this.temperature,
    this.notes,
    this.createdAt,
  });

  factory TriageRecord.fromJson(Map<String, dynamic> json) {
    return TriageRecord(
      id: json['id'] as int?,
      sosEventId: json['sosEventId'] as int? ?? 0,
      heartRate: json['heartRate'] as int?,
      systolicBp: json['systolicBp'] as int?,
      diastolicBp: json['diastolicBp'] as int?,
      spo2: json['spo2'] as int?,
      temperature: (json['temperature'] as num?)?.toDouble(),
      notes: json['notes'] as String?,
      createdAt: json['createdAt'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'sosEventId': sosEventId,
      if (heartRate != null) 'heartRate': heartRate,
      if (systolicBp != null) 'systolicBp': systolicBp,
      if (diastolicBp != null) 'diastolicBp': diastolicBp,
      if (spo2 != null) 'spo2': spo2,
      if (temperature != null) 'temperature': temperature,
      if (notes != null) 'notes': notes,
    };
  }
}
